src/main/java/com/iuc/requests/controller/UserController.java

```diff
@@ -36,7 +46,155 @@ public StaffDto findStaffByMatricule(@RequestParam(value = "userRegistration") S
     }
 
     @PostMapping("/staff")
-    public StaffDto save(@RequestBody StaffDto staffDto) {
-        return userService.save(staffDto);
+    public StaffDto saveStaff(@RequestBody StaffDto staffDto) {
+        return userService.saveStaff(staffDto);
+    }
+
+
+    @DeleteMapping("/staff")
+    public void deleteStaffByMatricule(@RequestParam(value = "userRegistration") String userRegistration){
+
+        if(!userService.findStaffByMatricule(userRegistration).equals(null)){
```

Comment:
I think don't need a condition.
Just get the existing staff for deleting by ``Staff staff = userService.findStaffByMatricule()`` Then send the entity to service for deleting.

Reply:
The comment has been taken into consideration